whois-commons/src/main/java/net/ripe/db/whois/common/WhoisAWSPropertyResolver.java

```diff
@@ -0,0 +1,77 @@
+package net.ripe.db.whois.common;
+
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagement;
+import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClient;
+import com.amazonaws.services.simplesystemsmanagement.model.GetParametersByPathRequest;
+import com.amazonaws.services.simplesystemsmanagement.model.GetParametersByPathResult;
+import com.amazonaws.services.simplesystemsmanagement.model.Parameter;
+import net.ripe.db.whois.common.profiles.WhoisProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+@Configuration
+@Profile(WhoisProfile.AWS_DEPLOYED)
+@PropertySource(value = "classpath:version.properties", ignoreResourceNotFound = true)
+public class WhoisAWSPropertyResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WhoisAWSPropertyResolver.class);
+
+    //TODO: get region to deploy from gitlab
+    private static final String REGION = "eu-central-1";
+
+    @Bean
+    public static PropertySourcesPlaceholderConfigurer properties(){
+        LOGGER.info("AWS profile : using ssm parameter store ");
+
+        Properties properties = new Properties();
+        properties.putAll( getParametersByEnvAndApp("/whois/", true));
+
+        PropertySourcesPlaceholderConfigurer propertySourceConfig = new PropertySourcesPlaceholderConfigurer();
+        propertySourceConfig.setProperties(properties);
+
+        return propertySourceConfig;
+    }
+
+    public static Map<String, Object> getParametersByEnvAndApp(final String path, final boolean encryption) {
```

Comment:
If encryption is only ever true, why pass it as an argument?

Reply:
ok, I will make it default true always, as it needs to decrypt the passwords